src/Nethermind/Nethermind.Crypto/KzgPolynomialCommitments.cs

```diff
@@ -51,16 +73,29 @@ public static bool TryComputeCommitmentV1(ReadOnlySpan<byte> commitment, Span<by
         return false;
     }
 
-    public static unsafe bool VerifyProof(ReadOnlySpan<byte> commitment, ReadOnlySpan<byte> z, ReadOnlySpan<byte> y, ReadOnlySpan<byte> proof)
+    public static bool VerifyProof(ReadOnlySpan<byte> commitment, ReadOnlySpan<byte> z, ReadOnlySpan<byte> y,
+        ReadOnlySpan<byte> proof)
     {
-        if (_ckzgSetup == IntPtr.Zero)
+        try
+        {
+            return Ckzg.Ckzg.VerifyKzgProof(commitment, z, y, proof, _ckzgSetup);
+        }
+        catch (Exception e) when (e is ArgumentException or ApplicationException or InsufficientMemoryException)
         {
-            throw new InvalidOperationException("KZG is not initialized");
+            return false;
         }
+    }
 
-        fixed (byte* commitmentPtr = commitment, zPtr = z, yPtr = y, proofPtr = proof)
+    public static bool AreProofsValid(byte[] blobs, byte[] commitments, byte[] proofs)
+    {
+        try
+        {
+            return Ckzg.Ckzg.VerifyBlobKzgProofBatch(blobs, commitments, proofs, blobs.Length / Ckzg.Ckzg.BytesPerBlob,
+                _ckzgSetup);
+        }
+        catch (Exception e) when (e is ArgumentException or ApplicationException or InsufficientMemoryException)
         {
-            return Ckzg.Ckzg.VerifyKzgProof(commitmentPtr, zPtr, yPtr, proofPtr, _ckzgSetup) == 0;
+            return false;
```

Comment:
Same question as for `VerifyProof`

Reply:
Should not break a block

Reply:
(but heavy, taking milliseconds, we need to do something about that, added to TODO)